From 971945b6aa3d312bc4a9011c002a9ba7e2b9c22f Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Thu, 10 Aug 2023 17:32:58 +0200
Subject: [PATCH 058/120] memory: stm32-omi: Remove stm32_omi_dlyb_stop()
 return value

No need to test regmap_write() return value, so stm32_omi_dlyb_stop()
return value can be remove.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git b83931d0f4f59d2864c0eb68e669fec6e05de772 memory: stm32-omi: Remove stm32_omi_dlyb_stop() return value]
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: I14283228e6f20f566747641323a184a84fe165a4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/322457
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/memory/stm32-omi.c | 20 ++++++--------------
 drivers/spi/stm32_ospi.c   | 17 +++++------------
 include/stm32_omi.h        |  2 +-
 3 files changed, 12 insertions(+), 27 deletions(-)

diff --git a/drivers/memory/stm32-omi.c b/drivers/memory/stm32-omi.c
index c4f94065e00..ce8ebafcae2 100644
--- a/drivers/memory/stm32-omi.c
+++ b/drivers/memory/stm32-omi.c
@@ -221,20 +221,14 @@ static u32 stm32_omi_find_byp_cmd(u16 period_ps)
 		return FIELD_PREP(DLYBOS_BYP_CMD_MASK, max + 1);
 }
 
-int stm32_omi_dlyb_stop(struct udevice *dev)
+void stm32_omi_dlyb_stop(struct udevice *dev)
 {
 	struct stm32_omi_plat *omi_plat = dev_get_plat(dev);
-	int ret;
 
 	/* disable delay block */
-	ret = regmap_write(omi_plat->regmap,
-			   omi_plat->dlyb_base + SYSCFG_DLYBOS_CR,
-			   0x0);
-
-	if (ret)
-		dev_err(dev, "Error when stopping delay block\n");
-
-	return ret;
+	regmap_write(omi_plat->regmap,
+		     omi_plat->dlyb_base + SYSCFG_DLYBOS_CR,
+		     0x0);
 }
 
 int stm32_omi_dlyb_configure(struct udevice *dev,
@@ -242,7 +236,7 @@ int stm32_omi_dlyb_configure(struct udevice *dev,
 {
 	struct stm32_omi_plat *omi_plat = dev_get_plat(dev);
 	u32 sr, mask;
-	int ret, err;
+	int ret;
 
 	if (!omi_plat->regmap || !omi_plat->dlyb_base)
 		return -EINVAL;
@@ -267,9 +261,7 @@ int stm32_omi_dlyb_configure(struct udevice *dev,
 				       STM32_DLYBOS_TIMEOUT_MS);
 	if (ret) {
 		dev_err(dev, "Delay Block lock timeout\n");
-		err = stm32_omi_dlyb_stop(dev);
-		if (err)
-			return err;
+		stm32_omi_dlyb_stop(dev);
 	}
 
 	return ret;
diff --git a/drivers/spi/stm32_ospi.c b/drivers/spi/stm32_ospi.c
index 18ac48100d5..db06818b3eb 100644
--- a/drivers/spi/stm32_ospi.c
+++ b/drivers/spi/stm32_ospi.c
@@ -316,9 +316,7 @@ static int stm32_ospi_calibration(struct udevice *bus, uint freq)
 	if (!ret_tcr0)
 		stm32_omi_dlyb_get_cr(priv->omi_dev, &dlyb_cr);
 
-	ret = stm32_omi_dlyb_stop(priv->omi_dev);
-	if (ret)
-		return ret;
+	stm32_omi_dlyb_stop(priv->omi_dev);
 
 	ret = stm32_omi_dlyb_configure(priv->omi_dev, false, 0);
 	if (ret)
@@ -335,9 +333,7 @@ static int stm32_ospi_calibration(struct udevice *bus, uint freq)
 	if (window_len_tcr0 >= window_len_tcr1) {
 		clrbits_le32(regs_base + OSPI_TCR, OSPI_TCR_SSHIFT);
 
-		ret = stm32_omi_dlyb_stop(priv->omi_dev);
-		if (ret)
-			return ret;
+		stm32_omi_dlyb_stop(priv->omi_dev);
 
 		ret = stm32_omi_dlyb_set_cr(priv->omi_dev, dlyb_cr);
 		if (ret)
@@ -369,9 +365,7 @@ static int stm32_ospi_claim_bus(struct udevice *dev)
 
 	priv->cs_used = slave_cs;
 
-	ret = stm32_omi_dlyb_stop(priv->omi_dev);
-	if (ret)
-		return ret;
+	stm32_omi_dlyb_stop(priv->omi_dev);
 
 	/* Set chip select */
 	clrsetbits_le32(regs_base + OSPI_CR, OSPI_CR_CSSEL,
@@ -380,6 +374,7 @@ static int stm32_ospi_claim_bus(struct udevice *dev)
 
 	dcr2 = readl(regs_base + OSPI_DCR2);
 	prescaler = (dcr2 & OSPI_DCR2_PRESC_MASK) >> OSPI_DCR2_PRESC_SHIFT;
+	stm32_omi_dlyb_stop(priv->omi_dev);
 	bus_freq = ospi_clk / (prescaler + 1);
 
 	/* calibration needed above 50MHz */
@@ -389,9 +384,7 @@ static int stm32_ospi_claim_bus(struct udevice *dev)
 			dev_info(dev->parent, "Set flash frequency to a safe value (%d Hz)\n",
 				 STM32_DLYB_FREQ_THRESHOLD);
 
-			ret = stm32_omi_dlyb_stop(priv->omi_dev);
-			if (ret)
-				return ret;
+			stm32_omi_dlyb_stop(priv->omi_dev);
 
 			clrbits_le32(regs_base + OSPI_TCR, OSPI_TCR_SSHIFT);
 			ret = stm32_ospi_set_speed(dev->parent, STM32_DLYB_FREQ_THRESHOLD);
diff --git a/include/stm32_omi.h b/include/stm32_omi.h
index b35d21134b3..aef2c6f4d24 100644
--- a/include/stm32_omi.h
+++ b/include/stm32_omi.h
@@ -169,7 +169,7 @@ int stm32_omi_dlyb_configure(struct udevice *dev,
 int stm32_omi_dlyb_find_tap(struct udevice *dev, bool rx_only, u8 *window_len);
 int stm32_omi_dlyb_set_cr(struct udevice *dev, u32 dlyb_cr);
 void stm32_omi_dlyb_get_cr(struct udevice *dev, u32 *dlyb_cr);
-int stm32_omi_dlyb_stop(struct udevice *dev);
+void stm32_omi_dlyb_stop(struct udevice *dev);
 int stm32_omi_tx_poll(struct udevice *dev, u8 *buf, u32 len, bool read);
 int stm32_omi_wait_cmd(struct udevice *dev);
 int stm32_omi_wait_for_not_busy(struct udevice *dev);
-- 
2.47.2

